elisp/intero.el

```diff
@@ -903,8 +904,14 @@ STORE-PREVIOUS is non-nil, note the caller's buffer in
             (get-buffer name)
           (with-current-buffer
               (get-buffer-create name)
-            (cd root)
-            (intero-repl-mode)
+	    ;; The new buffer doesn't know if the initial buffer was hosted
+	    ;; remotely or not, so we need to extend by the host of the
+	    ;; initial buffer to cd. We could also achieve this by setting the
+	    ;; buffer's intero-buffer-host, but intero-repl-mode wipes this, so
+	    ;; we defer setting that until after.
+            (cd (intero-extend-path-by-buffer-host root initial-buffer))
+            (intero-repl-mode) ; wipes buffer-local variables
+	    (setq intero-buffer-host (intero-buffer-host initial-buffer))
```

Comment:
I'm not sure - it calls `intero-buffer-filename` in the background, and I wouldn't have thought the working directory is enough to provide that. `intero-repl-mode` blows it all away, in any case.